src/common/Form/index.jsx

```diff
@@ -1,63 +1,105 @@
-import React, { useState } from 'react'
-import { useDispatch } from 'react-redux';
+import { useState, useEffect } from 'react'
+import { useDispatch } from 'react-redux'
 import { useNavigate } from 'react-router-dom'
 
 import styles from './index.module.css'
 import { Button } from '..'
 import { sendMessageAction } from '../../actions'
+import { isEmailValid } from './utils'
+
+const initialUserState = {
+  email: '',
+  reason: '',
+  nick: '',
+  content: '',
+}
 
 function Form() {
   const dispatch = useDispatch()
   const navigate = useNavigate()
 
-  const [email, setEmail ] = useState('')
-  const [reason, setReason ] = useState('')
-  const [name, setName ] = useState('')
-  const [content, setContent ] = useState('')
+  const [user, setUser] = useState(initialUserState)
   const [checked, setChecked] = useState(true)
 
-  const handleInput = (e, fun)  => {
-    fun(e.target.value)
+  const handleChange = (e) => {
+    const name = e.target.name
+    const value = e.target.value
+    setUser({ ...user, [name]: value })
   }
 
   const handleSubmit = (e) => {
-    const message = {
+    e.preventDefault()
+    if (!isEmailValid(user.email)) {
+      return null
+    }
+    if (user.nick && user.reason && user.email && user.content) {
+      const message = {
         person: {
-          email: email,
-          name: name
+          email: user.email,
+          name: user.nick,
         },
-        message: content,
-        department: reason,
+        message: user.content,
+        department: user.reason,
         newsletter: checked,
+      }
```

Comment:
Here shoud be an else satement, asking for filling in all the fields.

Reply:
This is something that HTML does by itself with 'required'